### PR TITLE
🎉 Source Github: PullRequestCommentReactions - re-implemented using GraphQL

### LINKED ISSUE #14408: 🎉 Source Github: re-implement `Reactions` streams using GraphQL API
# Problem
Source GitHub `Reactions` streams are still not very fast because they nested and require a lot of HTTP requests 
to sync all data.

For example to sync stream: `pull_request_comment_reactions` we need to iterate over comments
and for every comment make separate HTTP request to get a list of reaction for specific comment.
It requires a lot of HTTP requests - it's slow

# Solution
One solution we can use is to re-implement REST API -> GraphQL API.
It allows to reduce number of HTTP requests.
One GraphQL API HTTP request can fetch a lot of reactions for a lot of comments.

We can re-implement all Reactions streams using GraphQL API to improve sync performance.

GraphQL API still not silver bullet: it reduce number of HTTP requests but it's CPU intensive
and GitHub count total cost of such GraphQL requests. If you spend all hourly costs you need to wait.

### PR DESCRIPTION
**Pull request by grubberr**
Signed-off-by: Sergey Chvalyuk <grubberr@gmail.com>

## Performance for stream PullRequestCommentReactions

Type        | HTTP requests | Sync time           | Number of records  
--------- |---------------  |-----------------|-------------------
GraphQL | 2796                  | 674 seconds      | 2365
REST API | 28671                | 17068 seconds | 2365

Some notes:

- Traversing Graph algorithm produce UNSORTED records (not fully sorted)

## How
*Describe the solution*

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [x] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [x] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [x] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [x] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [x] Create a non-forked branch based on this PR and test the below items on it
- [x] Build is successful
- [x] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [x] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

*Put your acceptance tests output here.*

</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-github/source_github/graphql.py [hunk 1]
```diff
@@ -87,12 +93,113 @@ def get_query_reviews(owner, name, first, after, number=None):
...
+
+
+class QueryReactions:
+
+    AVERAGE_REVIEWS = 5
+    AVERAGE_COMMENTS = 2
+    AVERAGE_REACTIONS = 2
+
+    def get_query_root_repository(self, owner, name, first, after=None):
+        op = sgqlc.operation.Operation(_schema_root.query_type)
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-github/source_github/graphql.py:107]: I see we use this line in a lot of methods. May be better shift it to class or instance attribute.

**Review by lazebnyi — CHANGES_REQUESTED** [review]

**Review by lazebnyi — APPROVED** [review]

File: airbyte-integrations/connectors/source-github/source_github/graphql.py [hunk 2]
```diff
@@ -87,12 +93,116 @@ def get_query_reviews(owner, name, first, after, number=None):
...
-    )
+    select_user_fields(user)
     return str(op)
+
+
+class QueryReactions:
+
+    AVERAGE_REVIEWS = 5
+    AVERAGE_COMMENTS = 2
+    AVERAGE_REACTIONS = 2
```
Comment (Phlair) [inline airbyte-integrations/connectors/source-github/source_github/graphql.py:104]: Unclear what these numbers mean and where they come from... code comment would be good.

File: airbyte-integrations/connectors/source-github/source_github/graphql.py [hunk 3]
```diff
@@ -87,12 +93,116 @@ def get_query_reviews(owner, name, first, after, number=None):
...
+
+    def get_query_root_repository(self, owner, name, first, after=None):
```
Comment (Phlair) [inline airbyte-integrations/connectors/source-github/source_github/graphql.py:106]: all these funcs could do with:

- type hints on params
- brief docstring explaining what they are achieving.

**Review by Phlair — APPROVED** [review]
Approving to not block but would appreciate adding docstrings & code comments to make the logic clear for any dev that works on this in the future.
Great move for optimisation though 👍

File: airbyte-integrations/connectors/source-github/source_github/graphql.py (hunk 3 above)
Comment (grubberr) [reply airbyte-integrations/connectors/source-github/source_github/graphql.py]: added

**Pull request merged**